src/main/java/edu/cornell/kfs/fp/batch/service/impl/CreateAccountingDocumentReportSerivceImpl.java

```diff
@@ -0,0 +1,162 @@
+                    CuFPKeyConstants.REPORT_CREATE_ACCOUNTING_DOCUMENT_ERRED_DOCUMENTS_NONE));
+        }
+    }
+    
+    private void generateSuccessDocumentSection(List<CreateAccounntingDocumentReportItemDetail> successDocuments) {
+        reportWriterService.writeSubTitle(configurationService.getPropertyValueAsString(
+                CuFPKeyConstants.REPORT_CREATE_ACCOUNTING_DOCUMENT_ROUTED_DOCUMENTS_SUB_HEADER));
+        if (CollectionUtils.isNotEmpty(successDocuments)) {
+            successDocuments.stream().forEach(detail -> generateSuccessDetail(detail));
+        } else {
+            reportWriterService.writeFormattedMessageLine(configurationService.getPropertyValueAsString(
+                    CuFPKeyConstants.REPORT_CREATE_ACCOUNTING_DOCUMENT_ROUTED_DOCUMENTS_NONE));
+        }
+    }
+    
+    private void generateErrorDetail(CreateAccounntingDocumentReportItemDetail detail) {
+        generateSharedDetails(detail);
+        reportWriterService.writeFormattedMessageLine(fomratString(configurationService.getPropertyValueAsString(
+                CuFPKeyConstants.REPORT_CREATE_ACCOUNTING_DOCUMENT_ERRED_DOCUMENTS_MESSAGE), detail.getErrorMessage()));
+        reportWriterService.writeNewLines(1);
+    }
+    
+    private void generateSuccessDetail(CreateAccounntingDocumentReportItemDetail detail) {
+        generateSharedDetails(detail);
+        reportWriterService.writeFormattedMessageLine(fomratString(configurationService.getPropertyValueAsString(
+                CuFPKeyConstants.REPORT_CREATE_ACCOUNTING_DOCUMENT_ROUTED_DOCUMENTS_DOCUMENT_NUMBER), detail.getDocumentNumber()));
+        reportWriterService.writeNewLines(1);
+    }
+    
+    private void generateSharedDetails(CreateAccounntingDocumentReportItemDetail detail) {
+        reportWriterService.writeFormattedMessageLine(fomratString(configurationService.getPropertyValueAsString(
+                CuFPKeyConstants.REPORT_CREATE_ACCOUNTING_DOCUMENT_SHARED_INDEX_NUMBER), detail.getIndexNumber()));
+        reportWriterService.writeFormattedMessageLine(fomratString(configurationService.getPropertyValueAsString(
+                CuFPKeyConstants.REPORT_CREATE_ACCOUNTING_DOCUMENT_SHARED_DOCUMENT_TYPE), detail.getDocumentType()));
+        reportWriterService.writeFormattedMessageLine(fomratString(configurationService.getPropertyValueAsString(
+                CuFPKeyConstants.REPORT_CREATE_ACCOUNTING_DOCUMENT_SHARED_DOCUMENT_DESCRIPTION), detail.getDocumentDescription()));
+        reportWriterService.writeFormattedMessageLine(fomratString(configurationService.getPropertyValueAsString(
+                CuFPKeyConstants.REPORT_CREATE_ACCOUNTING_DOCUMENT_SHARED_DOCUMENT_EXPLANATION), detail.getDocumentExplanation()));
+    }
+    
+    public String fomratString(String pattern, Object... args) {
+        return MessageFormat.format(pattern, args);
+    }
+    
+    @Override
+    public void sendReportEmail(String toAddress, String fromAddress) {
```

Comment:
There is duplicate code in file CorporateBilledCorporatePaidRouteStepReportServiceImpl.java. Would it be 
feasible/desirable to refactor this method into another (static) class such as CuBatchFileUtils?